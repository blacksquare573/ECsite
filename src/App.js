import "./App.css";
import CategoriesMenu from "./components/Categories/CategoriesMenu";

function App() {
  return (
    <div className="App">
      <CategoriesMenu />
    </div>
  );
}

export default App;
