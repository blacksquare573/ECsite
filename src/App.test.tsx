/**
 * @jest-environment jsdom
 */

import React from "react";
import { render, screen } from "@testing-library/react";
import { jest } from "@jest/globals";
import SkuTitle from "./components/SkuPage/SkuTitle.component";
import { SkuTitleProps } from "./components/SkuPage/SkuTitle.component";
import renderer from "react-test-renderer";
import App from "./App";

// test("renders learn react link", () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

test("render skuTitle", () => {
  const mockSkuName = "両面使える敷きパッド　シングル(NクールWSP n-s GY S)";
  const mockSkuId = 10011;
  expect.assertions(1);
  const testSkuTitle = renderer
    .create(<SkuTitle skuName={mockSkuName} skuId={mockSkuId} />)
    .toJSON();
  return expect(testSkuTitle).toMatchSnapshot();
});

test("render App", () => {
  render(<App />);
});
