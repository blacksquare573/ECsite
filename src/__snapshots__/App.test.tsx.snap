// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render skuTitle 1`] = `
Array [
  <div
    className="sku-name-container"
  >
    <h1
      className="sku-name"
    >
      両面使える敷きパッド　シングル(NクールWSP n-s GY S)
    </h1>
  </div>,
  <div
    className="sku-code-and-score-container"
  >
    <div
      className="sku-code-container"
    >
      <p
        className="sku-code"
      >
        商品コード 
        10011
      </p>
    </div>
    <div
      className="sku-score-container"
    >
      <p
        className="sku-score"
      >
        <span
          className="score-stars"
        >
          平均評価4.1点
        </span>
        <span
          className="score-reviews"
        >
          （97）
        </span>
      </p>
    </div>
  </div>,
]
`;
